purchases/src/main/kotlin/com/revenuecat/purchases/common/Dispatcher.kt

```diff
@@ -59,28 +62,25 @@ internal open class Dispatcher(
     ) {
         synchronized(this.executorService) {
             if (!executorService.isShutdown) {
-                val future = if (delay != Delay.NONE && executorService is ScheduledExecutorService) {
+                val commandHandlingExceptions = Runnable {
+                    try {
+                        command.run()
+                    } catch (@Suppress("TooGenericExceptionCaught") e: Exception) {
+                        errorLog("Exception running command: $e")
+                        mainHandler?.post {
+                            throw e.cause ?: e
```

Comment:
can you explain this? why not just throw e?

Reply:
Probably not needed, I was mostly copying here what we had previously... In my tests, the cause was sometimes null, so I fallback to the original exception, but we could probably just propagate the original exception

Reply:
Yes, I am not sure why we were doing `e.cause` 🤔